core/server/middleware/theme-handler.js

```diff
@@ -60,6 +60,8 @@ themeHandler = {
 
         // clear the view cache
         blogApp.cache = {};
+        // reset the asset hash
+        config.assetHash = null;
```

Comment:
The important detail is that all middleware theme activation/refresh/cache clear code assumes the theme name changes. In case 3 it doesn't.